src/main/java/com/amazonaws/services/kinesis/clientlibrary/proxies/KinesisProxy.java

```diff
@@ -393,9 +468,12 @@ public Shard getShard(String shardId) {
                 }
             } while (response.getStreamDescription().isHasMoreShards());
         }
-        this.listOfShardsSinceLastGet.set(shardIterationState.getShards());
+        List<Shard> shards = shardIterationState.getShards();
+        this.cachedShardMap = shards.stream().collect(Collectors.toMap(Shard::getShardId, Function.identity()));
+        this.lastCacheUpdateTime = Instant.now();
```

Comment:
Can this be a separate method. The method says getShardList, lets scope it to doing just that.

Reply:
The caching behavior allows the getShard to effectively piggy back on getShardList.  We really want that to prevent additional scans of the shard map.  We also can't change the method as it's part of the interface.